scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/ExtensionMethods.scala

```diff
@@ -3,23 +3,22 @@ package pl.touk.nussknacker.engine.extension
 import pl.touk.nussknacker.engine.definition.clazz.{ClassDefinitionSet, MethodDefinition}
 import pl.touk.nussknacker.engine.extension.ExtensionMethods.extensionMethodsHandlers
 
-import java.lang.reflect.{Method, Modifier}
+import java.lang.reflect.{InvocationTargetException, Method, Modifier}
+import scala.util.{Failure, Try}
 
 class ExtensionsAwareMethodInvoker(classDefinitionSet: ClassDefinitionSet) {
 
   private val toInvocationTargetConvertersByClass =
     extensionMethodsHandlers
       .map(e => e.invocationTargetClass -> e.createConverter(classDefinitionSet))
-      .toMap[Class[_], ToExtensionMethodInvocationTargetConverter[_]]
+      .toMap[Class[_], ToExtensionMethodInvocationTargetConverter[_ <: ExtensionMethodInvocationTarget]]
 
   def invoke(method: Method)(target: Any, arguments: Array[Object]): Any = {
     if (toInvocationTargetConvertersByClass.contains(method.getDeclaringClass)) {
       toInvocationTargetConvertersByClass
         .get(method.getDeclaringClass)
         .map(_.toInvocationTarget(target))
-        // Maybe in future we could write some mechanism to invoke extension methods statically. What I mean is to
-        // find correct extension based on target and then implement simple switch to fire method based on name
-        .map(impl => method.invoke(impl, arguments: _*))
+        .map(impl => invokeMethodStatically(method, arguments, impl))
```

Comment:
Not exactly. Spring needs java.lang.reflect.Method to fit parsed method to an exact method (org.springframework.expression.spel.support.ReflectiveMethodResolver#resolve) because of it we need to have "invocation target" with defined methods.

Reply:
I guess the best solution would be to write an macro which will generate such invoke method. If a have some I will try to write sth like this

Reply:
Ahh I see, it's for method calling, not property executor. We could handled it otherwise. Currently, we added handling of extension methods inside `ReflectiveMethodExecutor`. Instead of this, we could just add another `MethodResolver`. Take a look that it is a very simple interface.

Reply:
Adding custom resolver is a good idea because it will simplify and isolate execution and discovery of extension methods but it will not resolve the problem of mapping method name to actual method or I don't have idea how to implement such behaviour without a such mappings

Reply:
There is also drawback that only ReflecitveExecutors are cached (org.springframework.expression.spel.ast.MethodReference#getCachedExecutor), so in a custom resolver we should also implement caching.

Reply:
> it will not resolve the problem of mapping method name to actual method or I don't have idea how to implement such behaviour without a such mappings

I thought about such a solution:
```
  private val IsClassRegex = "is(\\w+)".r("className")

  def invoke(methodName: String, arguments: Array[Object]): Any = methodName match {
    case IsClassRegex(`className`)  => convertEither(target).isRight
      (...)
    case _              => throw new IllegalAccessException(s"Cannot find method with name: '$methodName'")
  }
``` 

in a parent class of all conversions. Does it make sense? It is only a pseudo code. We can reduce complexity of finding the concrete `convertEither` in each method call from `O(n * m)` where n - count of converters, m - count of methods available in each converter, to `O(1)` by introducing the memoization

Reply:
>There is also drawback that only ReflecitveExecutors are cached (org.springframework.expression.spel.ast.MethodReference#getCachedExecutor), so in a custom resolver we should also implement caching.

In our case it won't be a problem, because in our `MethodResolver` we could have one `MethodExecutor` for every invocation. But I see the worse problem. If you have >1 revolver, the ReflectiveMethodResolver`s executor is not cashed :-/  We can work around this by having chain of responsibility inside the ReflectiveMethodResolver.

Reply:
Done